testing/test-helper.js

```diff
@@ -45,6 +45,27 @@ export function stubServiceForDoc(sandbox, ampdoc, serviceId, method) {
 }
 
 /**
+ * Resolves a promise when the method stub is called for the first time.
+ */
+export function stubWithCalledPromise(sandbox, object, method, opt_callMethod) {
+  const original = object[method];
+  let stub;
+  const promise = new Promise(resolve => {
+    stub = sandbox.stub();
+    const spy = sandbox.stub(object, method, () => {
+      const result = stub.apply(object, spy.lastCall.args);
+      if (opt_callMethod) {
+        original.apply(object, spy.lastCall.args);
+      }
+
+      resolve({value: result});
```

Comment:
If `result` is a promise, passing it to `resolve` will continue the promise chain. The test will hang if a test stubs a method and changes the stub behavior to return a promise, and the test doesn't immediately resolve the promise. I'm open to alternative solutions. See https://github.com/ampproject/amphtml/pull/9478/files#diff-f6081f6237f5cc3def355c9a96a881cdR718